cmd/dcrtime_fsck/dcrtime_fsck.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/decred/dcrd/dcrutil"
+	"github.com/decred/dcrtime/dcrtimed/backend"
+	"github.com/decred/dcrtime/dcrtimed/backend/filesystem"
+)
+
+var (
+	defaultHomeDir = dcrutil.AppDataDir("dcrtimed", false)
+	testnet        = flag.Bool("testnet", false, "Use testnet port")
+	fix            = flag.Bool("fix", false, "Try to correct correctable failures")
+	fsck           = flag.Bool("fsck", false, "Verify database content against blockchain")
+	verbose        = flag.Bool("v", false, "Print more information during run")
+	printHashes    = flag.Bool("printhashes", false, "Print all hashes")
+	fsRoot         = flag.String("source", defaultHomeDir, "Source directory")
+)
+
+func _main() error {
+	flag.Parse()
+
+	// Fsck
+	var root string
+	// XXX this needs to come out of chaincfg
+	if *testnet {
+		root = filepath.Join(*fsRoot, "data", "testnet3")
+	} else {
+		root = filepath.Join(*fsRoot, "data", "mainnet")
+	}
+
+	fmt.Printf("=== Root: %v\n", root)
+
+	fs, err := filesystem.NewDump(root)
+	if err != nil {
+		return err
+	}
+	defer fs.Close()
+
+	return fs.Fsck(&backend.FsckOptions{
+		Verbose:     *verbose,
+		PrintHashes: *printHashes,
+		Fix:         *fix,
```

Comment:
Fix is never actually referenced in the fsck code.  Errors get fixed regardless of whether this is true or false.

Reply:
Also got lost somehow. Adding it back.

Reply:
Resolved.